src/main/java/bio/terra/stairway/StairwayInstanceDao.java

```diff
@@ -0,0 +1,191 @@
+package bio.terra.stairway;
+
+import static bio.terra.stairway.DbUtils.commitTransaction;
+import static bio.terra.stairway.DbUtils.startReadOnlyTransaction;
+import static bio.terra.stairway.DbUtils.startTransaction;
+
+import bio.terra.stairway.exception.DatabaseOperationException;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.sql.DataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Database operations on the Stairway instance table */
+@SuppressFBWarnings(
+    value = "RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE",
+    justification = "Spotbugs doesn't understand resource try construct")
+public class StairwayInstanceDao {
```

Comment:
ditto does this class need to be public or just package private?

Reply:
I made a pass through all of the classes to clean up public classes that shouldn't be. Thanks for picking up on that.